src/java_bytecode/java_object_factory.cpp

```diff
@@ -1028,6 +1030,25 @@ void java_object_factoryt::gen_nondet_struct_init(
         substruct_in_place);
     }
   }
+
+  // If <class_identifier>.cproverValidate() can be found in the
+  // symbol table, we add a call:
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // expr.cproverValidate();
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  const irep_idt function_name =
+    "java::" + as_string(class_identifier) + ".cproverValidate:()V";
+  if(symbol_table.has_symbol(function_name))
+  {
+    code_function_callt fun_call;
+    fun_call.function() = symbol_exprt(function_name);
+    fun_call.arguments().push_back(address_of_exprt(expr));
+    code_typet fun_type;
+    fun_type.return_type() = void_type();
```

Comment:
The type should at least mention the `this` parameter. Get the type from the symbol table entry (`symbol_table.lookup(function_name).symbol_expr()`) instead of making it up here.